core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -359,4 +435,69 @@ class SpendReportingService(
         case ex: Exception =>
           Future.failed(RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, ex)))
       }
+
+//  def getSpendForUserWorkspaces(
+//    start: DateTime,
+//    end: DateTime
+//  ): Future[SpendReportingResults] =
+//    for {
+//      workspaces <- getOwnerWorkspaces()
+//      billing <- getBillingForWorkspaces(workspaces)
+//      query = getAllUserWorkspaceQuery(billing)
+//
+//    } yield query
+
+  def getSpendForAllWorkspaces(
+    project: RawlsBillingProjectName,
+    start: DateTime,
+    end: DateTime,
+    aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] = {
+    validateReportParameters(start, end)
+    requireProjectAction(project, SamBillingProjectActions.readSpendReport) {
+      for {
+        spendExportConf <- getSpendExportConfiguration(project)
+        projectNames <- getWorkspaceGoogleProjects(project)
+
+        query = getQuery(aggregations, spendExportConf)
+        queryJob = setUpQuery(query, spendExportConf, start, end, projectNames)
+
+        job: Job <- bigQueryService.use(_.runJob(queryJob)).unsafeToFuture().map(_.waitFor())
+        _ = logSpendQueryStats(job.getStatistics[JobStatistics.QueryStatistics])
+        result = job.getQueryResults()
+      } yield result.getValues.asScala.toList match {
+        case Nil =>
+          throw RawlsExceptionWithErrorReport(
+            StatusCodes.NotFound,
+            s"no spend data found for billing project ${project.value} between dates ${toISODateString(start)} and ${toISODateString(end)}"
+          )
+        case rows => extractSpendReportingResults(rows, start, end, projectNames, aggregations)
+      }
+    }
+  }
+
+  def getOwnerWorkspaces(): Future[Seq[WorkspaceListResponse]] =
+    workspaceServiceConstructor(ctx).listWorkspaces(WorkspaceFieldSpecs(), -1) map {
+      case JsArray(jsArray) =>
+        val workspaces = jsArray.map(_.convertTo[WorkspaceListResponse])
+        workspaces.filter(_.accessLevel == Owner)
+      case _ => throw new IllegalArgumentException("Expected a JsArray")
+    }
```

Comment:
Can this avoid using `listWorkspaces`? A couple reasons for this:
1. For many users, it will query for and calculate a lot more workspaces than you actually need; you're throwing away a lot of data
2. The `accessLevel` check is not ideal: https://github.com/broadinstitute/rawls/blob/995cd5571ca87abe6a25e23b9148be725e71912b/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L291-L293

edited to add: is this even used? I only see it called from a unit test.

Reply:
It will be used!  I'm just constructing helper methods piece by piece and making sure they work before I get to the part that uses it!

Reply:
I am not aware of a single function to do this. I think you'd have to call Sam to list the resources on which you have the appropriate action, and then use something like `WorkspaceRepository.listWorkspacesByIds` to get the details of those workspaces.

Reply:
Looking into this more closely, I can call `listWOrkspaces` with a param asking for only those the user owns.  That will save some re-calculation.  If I then check permissions correctly when getting the spend data, will I be safe?  Or would there be a situation where the user has permission to see the spend report for a billing project, but there's a workspace in that project they don't own and shouldn't see name and billing info on?